optimum/exporters/openvino/stateful.py

```diff
@@ -0,0 +1,234 @@
+#  Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import logging as log
+from typing import List
+
+import numpy as np
+from transformers import PretrainedConfig
+
+import openvino as ov
+from openvino.runtime import opset13
+from optimum.intel.utils.import_utils import _openvino_version, is_openvino_version
+from optimum.utils.normalized_config import NormalizedConfigManager
+
+
+def model_has_input_output_name(ov_model: ov.Model, name: str):
+    """
+    Helper function for checking that model has specified input or output name
+
+    Parameters:
+      ov_model (ov.Model):   # TODO: Can we derive the dimensions from the model topology?
+      name (str):
+          name of input or output
+
+    Returns:
+      True if input or output with requested name exists else False
+    """
+    return name in sum([list(t.get_names()) for t in ov_model.inputs + ov_model.outputs], [])
+
+
+def model_has_input(ov_model: ov.Model, name: str):
+    """
+    Helper function for checking that model has specified input name
+
+    Parameters:
+      ov_model (ov.Model):
+          opennvino model
+      name (str):
+          name of input
+
+    Returns:
+      True if input with requested name exists else False
+    """
+    return name in sum([list(t.get_names()) for t in ov_model.inputs], [])
+
+
+def model_has_cache_reorder(ov_model: ov.Model):
```

Comment:
I didn't find where the next two functions are used in the code. 
I also think that it would be better to have a property in the `model` object (e.g. `stateful: bool`) that will indicate whether the model is stateful or not.

Reply:
we already has stateful property in model, check for cache_reorder needed only in transformation and can be implemented in 1 line, so I removed helper function.

model_has_state - reused in OVModelForCausalLM constructor